classes/controllers/grid/issues/IssueGridHandler.inc.php

```diff
@@ -556,9 +562,17 @@ function unpublishIssue($args, $request) {
 			'issueIds' => $issue->getId(),
 			'count' => 5000, // large upper limit
 		]);
-		foreach ($submissionsIterator as $submission) {
+
+		foreach ($submissionsIterator as $submission) { /** @var Submission $submission */
 			$articleTombstoneManager->insertArticleTombstone($submission, $journal);
-			$submission = Services::get('submission')->edit($submission, ['status' => STATUS_QUEUED], $request);
+
+			$publications = $submission->getData('publications');
+			foreach ($publications as $publication) { /** @var Publication $publication */
+				if ($publication->getData('status') === STATUS_PUBLISHED && $publication->getData('issueId') === (int) $issue->getId()) {
+					$publication = Services::get('publication')->unpublish($publication);
+					$publication = Services::get('publication')->publish($publication);
```

Comment:
Maybe add a comment here explaining why the unpublish/publish thing is happening. Something like:

```
// Republish the publication in the issue, now that it's status has changed,
// to ensure the publication's status is restored to STATUS_SCHEDULED
// rather than STATUS_QUEUED
```